src/hooks.js

```diff
@@ -1,228 +1,84 @@
-  )
-
-  // Fetches and parses the JSON file of the typePaths we write at build time.
-  const fetchTypePaths = useCallback(async () => {
-    const req = await fetch(
-      `/${typePathsFilenamePrefix}${schemasDigest}.json`,
-      { headers: { 'Content-Type': 'application/json' } },
-    )
-
-    return await req.json()
-  }, [typePathsFilenamePrefix, schemasDigest])
-
-  // Normalizes preview data using browser-compatible normalize functions.
-  const normalizePreviewData = useCallback(
-    async rawPreviewData => {
-      const typePaths = await fetchTypePaths()
-
-      const nodeStore = new Map()
-      const createNode = node => nodeStore.set(node.id, node)
-      const hasNodeById = id => nodeStore.has(id)
-      const getNodeById = id => nodeStore.get(id)
-
-      const rootNodeId = await documentToNodes(rawPreviewData, {
-        typePaths,
-        createNode,
-        createNodeId,
-        createContentDigest,
-        hasNodeById,
-        getNodeById,
-        pluginOptions,
-        normalizeImageField,
-        normalizeLinkField,
-        normalizeSlicesField,
-        normalizeStructuredTextField,
-      })
-
-      const rootNode = nodeStore.get(rootNodeId)
-
-      const prefixedType = camelCase(rootNode.internal.type)
+  validateParameters,
+  getGlobalPluginOptions,
+  fetchTypePaths,
+  fetchPreviewData,
+  normalizePreviewData,
+} from './hook/helpers'
+
+export { mergePrismicPreviewData } from './hook/helpers'
+
+/**
+ * @typedef {Object} pluginOptions
+ * @property {string} repositoryName - Name of the Prismic repository to query.
+ * @property {string} accessToken - API token to query the Prismic API.
+ * @property {funcion} fetchLinks - Array of values that determines how Prismic fetches linked fields.
+ * @property {function} linkResolver - Function for Prismic to resolve links in the queried document.
+ *    @see {@link https://prismic.io/docs/javascript/beyond-the-api/link-resolving}
+ * @property {function} htmlSerializer - Function that allows Prismic to preprocess rich text fields.
+ *    @see {@link https://prismic.io/docs/javascript/beyond-the-api/html-serializer}
+ * @property {string} typePathsFilenamePrefix - Prefix to the typePaths json we generate at build time.
+ * @property {string} schemasDigest - Used for gatsby internals.
+ * @property {string} pathResolver - Function that allows for custom preview page path resolving.
+ */
+
+/**
+ * React hook providing preview data from Prismic identical in shape to the data
+ * created at build time. Images are not processed due to running in the browser.
+ * Instead, images reutrn their URL.
+ * @public
+ *
+ * @param {Object} location - Location object from @reach/router.
+ * @param {Object} pluginOptions - The {@link pluginOptions} for this preview.
+ *
+ * @returns An object containing normalized Prismic preview data directly from
+ *    the Prismic API.
+ */
+export const usePrismicPreview = (location, pluginOptions = {}) => {
```

Comment:
Could we change this file's name to `usePrismicPreview.js` instead of `hooks.js`? Since the file only contains the one hook. Should be easier to navigate through the files that way I think.

Reply:
File name has been changed (along with the `common/`, `browser/`, and `node/` structure we discussed.